### PR TITLE
Add support for AseDecimal

### PR DESCRIPTION
**Pull request by senseibaka**
Implements #10 
When the `UseAseDecimal` connection parameter is enabled, then the driver should use `AseDecimal` for all decimal representations.
Otherwise, regular `System.Decimal` will be used, and the existing `SqlDecimal` type should be used for to/from byte conversions.

### CODE REVIEW

File: src/AdoNetCore.AseClient/AseDecimal.cs [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+
+        /// <summary>
+        /// get the number of bytes required to represent the value (caller should add 1 byte to represent sign byte)
+        /// </summary>
+        public int BytesRequired
+        {
+            get
+            {
+                if (IsNull)
+                    throw new NullReferenceException();
```
Comment (sheikhjabootie) [inline src/AdoNetCore.AseClient/AseDecimal.cs]: @senseibaka  - seeing an explicit throw of NullReferenceException made me vomit a little in the back of my mouth.

File: src/AdoNetCore.AseClient/AseDecimal.cs [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+        public static bool TryParse(string s, out AseDecimal result)
+        {
+            try
+            {
+                result = Parse(s);
+                return true;
+            }
+            catch
+            {
+                result = new AseDecimal();
```
Comment (sheikhjabootie) [inline src/AdoNetCore.AseClient/AseDecimal.cs]: @senseibaka - Maybe this should be Zero?

File: src/AdoNetCore.AseClient/AseDecimal.cs [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+        {
+            return new AseDecimal(n.Backing.Floor());
+        }
+
+        public static AseDecimal Truncate(AseDecimal n, int outputScale)
+        {
+            if (outputScale < 0)
+            {
+                Logger.Instance?.WriteLine($"Invalid outputScale {outputScale}");
+                throw new AseException("Invalid value.", 30037);
```
Comment (sheikhjabootie) [inline src/AdoNetCore.AseClient/AseDecimal.cs]: @senseibaka - is this the same error code raised by the SAP version?

File: test/AdoNetCore.AseClient.Tests/Integration/HealthCheckTests.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 namespace AdoNetCore.AseClient.Tests.Integration
 {
-    [SetUpFixture]
+    //[SetUpFixture]
```
Comment (sheikhjabootie) [inline test/AdoNetCore.AseClient.Tests/Integration/HealthCheckTests.cs]: @senseibaka  - Is this meant to be checked-in. Why comment it out and not delete it? Or is it a mistake?

File: test/AdoNetCore.AseClient.Tests/Unit/LoginPacketTests.cs [hunk 5]
```diff
@@ -62,7 +61,25 @@ public void Matches_AdoNet4_Output()
...
             using (var ms = new MemoryStream())
             {
-                new LoginPacket("DES-PC0KFUSH", "USERNAME", "PASSWORD", "9360", "APPNAME", "", "us_english", "iso_1", "ADO.NET", 512, new CapabilityToken()).Write(ms, Encoding.ASCII);
+                new LoginPacket(
+                    "DES-PC0KFUSH",
```
Comment (sheikhjabootie) [inline test/AdoNetCore.AseClient.Tests/Unit/LoginPacketTests.cs]: @senseibaka - this name shouldn't be in here.

File: src/AdoNetCore.AseClient/AseDecimal.cs (hunk 1 above)
Comment (senseibaka) [reply src/AdoNetCore.AseClient/AseDecimal.cs]: @sheikhjabootie I've always wanted to do that 😄 🤢

File: src/AdoNetCore.AseClient/AseDecimal.cs (hunk 3 above)
Comment (senseibaka) [reply src/AdoNetCore.AseClient/AseDecimal.cs]: Yep

File: src/AdoNetCore.AseClient/AseDecimal.cs (hunk 2 above)
Comment (senseibaka) [reply src/AdoNetCore.AseClient/AseDecimal.cs]: Will change, though the caller shouldn't be using the result if `TryParse` returns `false` anyway

File: test/AdoNetCore.AseClient.Tests/Integration/HealthCheckTests.cs (hunk 4 above)
Comment (senseibaka) [reply test/AdoNetCore.AseClient.Tests/Integration/HealthCheckTests.cs]: Woops

File: test/AdoNetCore.AseClient.Tests/Unit/LoginPacketTests.cs (hunk 5 above)
Comment (senseibaka) [reply test/AdoNetCore.AseClient.Tests/Unit/LoginPacketTests.cs]: Woops

**Pull request merged**